ansible/roles/ocp4-workload-dil-agile-integration/tasks/provision_gogs.yaml

```diff
@@ -0,0 +1,143 @@
+  shell: "oc new-app \
+  -f  {{ ocp4_workload_dil_agile_integration_gogs_template_url }} \
+  -p GOGS_VERSION={{ ocp4_workload_dil_agile_integration_gogs_version }} \
+  -p SKIP_TLS_VERIFY={{ ocp4_workload_dil_agile_integration_gogs_skip_tls }} \
+  -p HOSTNAME=gogs.{{ route_subdomain }} \
+  -n {{ gogs_project }}"
+  register: create_gogs_app
+  failed_when: create_gogs_app.stderr != '' and 'already exists' not in create_gogs_app.stderr
+
+- name: Get required gogs postgre replicas
+  shell: "oc get dc/gogs-postgresql -o jsonpath='{.status.replicas}' -n {{ gogs_project }}"
+  failed_when: false
+  until: gogs_postgresql_replicas.stdout | int > 0
+  retries: 60
+  delay: 5
+  register: gogs_postgresql_replicas
+
+- name: wait for gogs-postgresql readiness
+  shell: "oc get dc/gogs-postgresql -o jsonpath='{.status.readyReplicas}' -n {{ gogs_project }}"
+  until: gogs_postgresql_available_replicas.stdout == gogs_postgresql_replicas.stdout
+  ignore_errors: yes
+  retries: 60
+  delay: 5
+  register: gogs_postgresql_available_replicas
+    
+- name: Get required gogs replicas
+  shell: "oc get dc/gogs -o jsonpath='{.status.replicas}' -n {{ gogs_project }}"
+  until: gogs_replicas.stdout | int > 0
+  retries: 60
+  delay: 5
+  failed_when: false
+  register: gogs_replicas
+
+- name: wait for gogs readiness
+  shell: "oc get dc/gogs -o jsonpath='{.status.readyReplicas}' -n {{ gogs_project }}"
+  until: gogs_available_replicas.stdout == gogs_replicas.stdout
+  ignore_errors: yes
+  retries: 60
+  delay: 5
+  register: gogs_available_replicas
+
+- name: Wait until Gogs API is available
+  uri: 
+    url: http://gogs.{{route_subdomain}}
+    method: GET
+    follow_redirects: safe
+    validate_certs: no
+  register: wait_gogs_result
+  until: wait_gogs_result is succeeded
+  ignore_errors: yes
+  retries: 15
+  delay: 30
+
+
+- name: Create Gogs users
+  uri:
+    url: http://gogs.{{ route_subdomain }}/user/sign_up
+    method: POST
+    follow_redirects: safe
+    return_content: yes
+    body_format: form-urlencoded
+    body:
+      user_name: 'user{{ item }}'
+      password: '{{ ocp4_workload_dil_agile_integration_gogs_user_password }}'
+      retype: '{{ ocp4_workload_dil_agile_integration_gogs_user_password }}'
+      email: 'user{{ item }}@example.com }}'
+    validate_certs: no
+    status_code: 200
+  register: create_tenant_result
+  ignore_errors: true
+  loop: "{{ range(1, num_users|int + 1, 1)|list }}"
+  when:
+    - num_users|int > 0
+
+- debug: var=create_tenant_result
+
+# - name: Get Git Repo
+#   uri:
+#     url: "{{ ocp4_workload_dil_agile_integration_gogs_git_config_url }}"
+#     return_content: yes
+#   register: get_gitrepo
+  
+# - debug: var=get_gitrepo.content
+
+- name: Config repo
+  uri: 
+    url: 'http://gogs.{{ route_subdomain }}/api/v1/repos/migrate'
+    method: POST
+    body: 
+      '{"clone_addr": "https://github.com/weimeilin79/dayinthelife-import.git", "uid": {{ item | int }},"repo_name": "dayintheliferepo"}'
```

Comment:
AgnosticD code should not depend on any repository owned by an individual. The repository should be managed by an organization such as https://github.com/redhat-cop/

I see that we can't use a dict here because of the nede to pass the `uid` as an int.